app/components/home/Challenge.tsx

```diff
@@ -29,213 +24,248 @@ export function Challenge({
   testID,
   currentUser,
 }: {
-  challengeDB: DBChallenge;
-  index: number;
-  firestoreCtrl: FirestoreCtrl;
-  navigation: any;
-  testID: string;
-  currentUser: DBUser;
+  readonly challengeDB: DBChallenge;
+  readonly index: number;
+  readonly firestoreCtrl: FirestoreCtrl;
+  readonly navigation: any;
+  readonly testID: string;
+  readonly currentUser: DBUser;
 }) {
-  const [isOpen, setIsOpen] = useState(false);
+  const [user, setUser] = useState<DBUser | null>(null);
   const [isLiked, setIsLiked] = useState(false);
   const [likes, setLikes] = useState<string[]>([]);
-  const [user, setUser] = useState<DBUser>();
+  const [comments, setComments] = useState<DBComment[]>([]);
 
-  const uri = "@/assets/images/no-image.svg";
+  // Double-tap logic
+  const [lastTap, setLastTap] = useState<number | null>(null);
 
-  // @ts-ignore - date is a Timestamp object
-  let challengeDate: Date = challengeDB.date ? challengeDB.date : new Date();
+  const placeholderImage = "https://via.placeholder.com/300";
 
   // Fetch user data
   useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        const userData = await firestoreCtrl.getUser(challengeDB.uid);
+        setUser(userData || null);
+      } catch (error) {
+        console.error("Error fetching user data:", error);
+      }
+    };
+
     if (challengeDB.uid) {
-      const fetchUser = async () => {
-        try {
-          const userData = await firestoreCtrl.getUser(challengeDB.uid);
-          setUser(userData);
-        } catch (error) {
-          console.error("Error fetching challenges: ", error);
-        }
-      };
       fetchUser();
     }
-  }, [user]);
+  }, [challengeDB.uid, firestoreCtrl]);
 
-  // Fetch likes data
+  // Fetch likes
   useEffect(() => {
```

Comment:
The like "switch" is not updated when you :
- Like the post on the home screen
- Maximize the post
- Remove the like

When coming to the home screen, the like should be removed.

Reply:
Right ! corrected that !